_minutes/2023-09-11-2023-09-14-tpac-extra.md

```diff
@@ -0,0 +1,520 @@
+     * [tomislav] But policy is not standardizable
+     * [devlin] Agreed.
+   * [tomislav] Custom element registry meeting
+     * https://www.w3.org/events/meetings/e5cc2d3f-6da6-4751-9417-18b80a524893/
+     * https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Scoped-Custom-Element-Registries.md
+     * [tomislav] Seems relevant to our interests. Would like at least one person to attend.
+     * [rob] Does Chrome keep track of which world created the dom element?
+     * [devlin] No
+     * [tomislav] I'm suggesting we ask them to keep track of the world.
+     * [rob] Not sure that's the best way to support custom elements. Could add a new method to the dom namespace to register.
+     * [tomislav] Could clash with the dom
+     * [rob] Yes, but content script can always clash with the dom.  …
+     * [devlin] If we introduce a dom.executeScript method, couldn't you just inject your custom element definition into the main world.
+     * [rob] dom.executeScript wouldn't cover this use case, because the global APIs may be unreliable due to web page's modifications to the main world. (..) but they shouldn't do that anyway.
+     * [devlin] … Seems like that might solve this.
+     * [tomislav] Does a custom element registered by a content script get exposed in the main world?
+     * [devlin] Only way to register a custom element is in the main world. Not sure I see the benefit of supporting this in the isolated world. Need compelling use cases.
+     * [rob] Firefox already supports exposing values across worlds through wrappers.
+     * [oliver] Another issue I've heard from developers is interacting with custom element properties e.g a custom element is registered in the main world and exposes a function, and the content script world calls it. But I think this is solved by having dom.executeScript and getting a handle on the element.
+     * [devlin] Yes. Not a handle though.
+     * [oliver] You could use document.querySelector.
+     * [devlin] Yes.
+     * [rob] You could pass a reference to the element via the args parameter.
+     * [devlin] Yuck.
+     * [rob] What's your concern with passing a reference to an element?
+     * [devlin] Complexity of mapping the JS variable to the DOM element in the isolated world, and then back to a JS handle in the main world.
+     * [rob] That should be safe.
+     * [devlin] yes, but there are other problems. … If in my JS I set data on an element, pass a reference to another world, and try to access that data in my other world scfript, would not see the data I set
+     * [rob] That is expected, a normal part of the isolated world concept.
+     * [devlin] Yep, exactly – but it's more surprising if I pass an element reference and get an element reference that those are totally different.  If I pass an ID and then re-fetch that element in the main world, it's more intuitive that they are different values.
```

Comment:
`id` or selector won't work because the element may be inside a closed shadow DOM for security. Can you use the already existing standard mechanism of passing DOM nodes via `new MouseEvent('click', {relatedTarget: myElem})`? And is there a better place to comment on this?

Reply:
Closed shadow roots are definitely tricky. The concern with using an existing mechanism like that was that it requires us to do quite a lot of magic behind the scenes to implement (and it's not a case of just re-using code, because we'd need to implement it for extension messaging). I don't think we have an issue here but it might be worth opening one cc/ @Rob--W

Reply:
This is for cross-world direct communication, not for the extension messaging.

Reply:
Resolving for now since we can discuss outside of this PR.